tests/e2e/service_invocation/service_invocation_test.go

```diff
@@ -89,27 +97,91 @@ var serviceinvocationTests = []struct {
 	},
 }
 
+var moreServiceinvocationTests = []struct {
+	in               string
+	remoteApp        string
+	appMethod        string
+	expectedResponse string
+}{
+	// For descriptions, see corresponding methods in dapr/tests/apps/service_invocation/app.go
+	{
+		"Test HTTP to HTTP",
+		"serviceinvocation-callee-1",
+		"httptohttptest",
+		"success",
+	},
+	{
+		"Test HTTP to gRPC",
+		"grpcapp",
+		"httptogrpctest",
+		"success",
+	},
+	{
+		"Test gRPC to HTTP",
+		"serviceinvocation-callee-1",
+		"grpctohttptest",
+		"success",
+	},
+	{
+		"Test gRPC to gRPC",
+		"grpcapp",
+		"grpctogrpctest",
+		"success",
+	},
+}
+
 func TestServiceInvocation(t *testing.T) {
 	externalURL := tr.Platform.AcquireAppExternalURL("serviceinvocation-caller")
 	require.NotEmpty(t, externalURL, "external URL must not be empty!")
-
+	var err error
 	// This initial probe makes the test wait a little bit longer when needed,
 	// making this test less flaky due to delays in the deployment.
-	_, err := utils.HTTPGetNTimes(externalURL, numHealthChecks)
+	_, err = utils.HTTPGetNTimes(externalURL, numHealthChecks)
 	require.NoError(t, err)
 
+	t.Logf("externalURL is '%s'\n", externalURL)
+
 	for _, tt := range serviceinvocationTests {
 		t.Run(tt.in, func(t *testing.T) {
 			body, err := json.Marshal(testCommandRequest{
 				RemoteApp: tt.remoteApp,
-				Method:    tt.method,
+				Method:    tt.appMethod,
+			})
+			require.NoError(t, err)
+
+			resp, err := utils.HTTPPost(
+				fmt.Sprintf("%s/tests/invoke_test", externalURL), body)
+			t.Log("checking err...")
+			require.NoError(t, err)
+
+			var appResp appResponse
+			t.Logf("unmarshalling..%s\n", string(resp))
+			err = json.Unmarshal(resp, &appResp)
+			require.NoError(t, err)
+			require.Equal(t, tt.expectedResponse, appResp.Message)
+		})
+	}
+
+	for _, tt := range moreServiceinvocationTests {
+		t.Run(tt.in, func(t *testing.T) {
+			body, err := json.Marshal(testCommandRequest{
+				RemoteApp: tt.remoteApp,
+				Method:    tt.appMethod,
 			})
 			require.NoError(t, err)
 
-			resp, err := utils.HTTPPost(fmt.Sprintf("%s/tests/invoke_test", externalURL), body)
+			url := fmt.Sprintf("http://%s/%s", externalURL, tt.appMethod)
+
+			t.Logf("url is '%s'\n", url)
```

Comment:
This code block is almost identical to the one above. Any reason they cannot be combined?

Reply:
The url has a different format here, and usage of appMethod is different.  In this case unlike above it really is the name of the method for the test to call (again not thru service invocation though).  I thought it was easier and more readable to keep them separate.